cmd/bundle/main.go

```diff
@@ -345,6 +365,134 @@ func dockerHubRepoDelete(token string, ref name.Reference) error {
 		return nil
 
 	default:
-		return fmt.Errorf("failed with HTTP status code %d: %s", resp.StatusCode, string(respData))
+		return fmt.Errorf("failed to delete image %q: %s (HTTP status code %d)",
+			ref.String(),
+			string(respData),
+			resp.StatusCode,
+		)
+	}
+}
+
+func icrLogin(apikey string) (string, string, error) {
+	data := fmt.Sprintf("grant_type=%s&apikey=%s",
+		url.QueryEscape("urn:ibm:params:oauth:grant-type:apikey"),
+		apikey,
+	)
+
+	req, err := http.NewRequest("POST", "https://iam.cloud.ibm.com/identity/token", strings.NewReader(data))
```

Comment:
Not sure what the best practice is in handling timeouts in golang. I see a NewRequestWithContext where one probably could pass in a context with a timeout. And http.Client has a timout ([here](https://pkg.go.dev/net/http#Client), but that's probably not set on the DefaultClient that you use ?